### PR TITLE
Improve @objc usage for objective-c usage/Xamarin bindings

### PR DESCRIPTION
**Pull request by nick-potts**
I was trying to use this for a Xamarin project but a lot of the bindings were missing. I've tried to add the required @objc bindings back in with the least invasive way possible.

### CODE REVIEW

File: Sources/Extensions/UIViewController+Hero.swift [hunk 1]
```diff
@@ -106,35 +69,87 @@ public extension UIViewController {
   }
 }
 
+//swift only
+extension UIViewController {
```
Comment (nick-potts) [inline Sources/Extensions/UIViewController+Hero.swift]: I'm sorry for having to split out these extensions like so, but objective c doesn't support these types of enums

**Comment by lkzhao** [comment]
@nick-potts Thanks for contributing. Supporting @objc and Xamarin is not our goal ATM. We will be slowly migrating away from objc and merging this in could potentially decrease the performance for pure Swift usage.
Instead of merging into the main project, how about I put up a link to your fork in the README for people who are using objective c and Xamarin? Does that sound good to you?

**Comment by lkzhao** [comment]
Also it might be hard to merge with @dDomovoj's PR https://github.com/lkzhao/Hero/pull/367 ATM.

**Comment by nick-potts** [comment]
I wouldn't link to it yet, I may end up changing quite a bit if it becomes a different fork given how many incompatibilities there are. If you are going the swift only route, I may end up rewriting it in Xamarin as a personal project.

**Comment by NashZhou** [comment]
@nick-potts It would be great if you could get it for C# devs. Is there any way to contact you if end up taking the task?

**Comment by nick-potts** [comment]
I'm gonna close this as its no longer relevant to myself and no progress was made.

**Pull request closed without merging**